packages/files/utils.js

```diff
@@ -0,0 +1,388 @@
+const busboy = require('busboy')
+const cuid = require('cuid')
+const { WriteStream } = require('fs-capacitor')
+const jwt = require('jsonwebtoken')
+const { validate: validateUUID } = require('uuid')
+const { z } = require('zod')
+
+const { CondoFile } = require('@open-condo/files/schema/utils/serverSchema')
+const FileAdapter = require('@open-condo/keystone/fileAdapter/fileAdapter')
+const { getKVClient } = require('@open-condo/keystone/kv')
+const { getLogger } = require('@open-condo/keystone/logging')
+
+const logger = getLogger('file-middleware')
+
+const AppClientSchema = z.object({
+    name: z.string().min(3),
+    modelNames: z.array(z.string()).min(1).optional(),
+    secret: z.string().min(8),
+}).strict()
+
+/** Top-level record: keys match /^[a-zA-Z0-9-]+$/, values are AppClientSchema */
+const AppClientsSchema = z.record(
+    z.string().regex(/^[a-zA-Z0-9-]+$/),
+    AppClientSchema
+)
+
+/** Validate and parse FILE_APP_CLIENTS. Returns the validated object. */
+function validateAndParseAppClients (data) {
+    return AppClientsSchema.parse(data)
+}
+
+// ---------------------- utilities ----------------------
+
+const FINGERPRINT_RE = /^[a-zA-Z0-9!#$%()*+-;=,:[\]/.?@^_`{|}~]{5,42}$/
+
+function sendError (res, status, message) {
```

Comment:
all you need is to call next with GQLError

Reply:
Do a massive refactor for that. Also change tests to check error format, but I need to add expressErrorHandler at the end of the my middleware, because it connected slightly different way